pipelines/pingcap/tidb-test/latest/ghpr_integration_common_test.groovy

```diff
@@ -0,0 +1,179 @@
+                    stage("Test") {
+                        options { timeout(time: 40, unit: 'MINUTES') }
+                        steps {
+                            dir('tidb') {
+                                cache(path: "./bin", filter: '**/*', key: "ws/${BUILD_TAG}/dependencies") {
+                                    sh label: "print version", script: """
+                                        pwd && ls -alh
+                                        ls bin/tidb-server && chmod +x bin/tidb-server && ./bin/tidb-server -V
+                                        ls bin/pd-server && chmod +x bin/pd-server && ./bin/pd-server -V
+                                        ls bin/tikv-server && chmod +x bin/tikv-server && ./bin/tikv-server -V
+                                    """
+                                }
+                            }
+                            dir('tidb-test') {
+                                cache(path: "./", filter: '**/*', key: "ws/tidb-test/rev-${REFS.pulls[0].sha}") {
+                                    sh """
+                                        mkdir -p bin
+                                        cp ${WORKSPACE}/tidb/bin/* bin/ && chmod +x bin/*
+                                        ls -alh bin/
+                                    """
+                                    container("golang") {
+                                        sh label: "test_params=${TEST_PARAMS} ", script: """
+                                            #!/usr/bin/env bash
+                                            params_array=(\${TEST_PARAMS})
+                                            TEST_DIR=\${params_array[0]}
+                                            TEST_SCRIPT=\${params_array[1]}
+                                            echo "TEST_DIR=\${TEST_DIR}"
+                                            echo "TEST_SCRIPT=\${TEST_SCRIPT}"
+                                            if [[ "${TEST_STORE}" == "tikv" ]]; then
+                                                echo '[storage]\nreserve-space = "0MB"'> tikv_config.toml
+                                                bash ${WORKSPACE}/scripts/pingcap/tidb-test/start_tikv.sh
+                                                export TIDB_SERVER_PATH="${WORKSPACE}/tidb-test/bin/tidb-server"
+                                                export TIKV_PATH="127.0.0.1:2379"
+                                                export TIDB_TEST_STORE_NAME="tikv"
+                                                cd \${TEST_DIR} && chmod +x *.sh && \${TEST_SCRIPT}
+                                            else
+                                                export TIDB_SERVER_PATH="${WORKSPACE}/tidb-test/bin/tidb-server"
+                                                export TIDB_TEST_STORE_NAME="unistore"
+                                                cd \${TEST_DIR} && chmod +x *.sh && \${TEST_SCRIPT}
+                                            fi
```

Comment:
Those lines should be wrapped in a script file and make call it with params `TEST_STORE` and `WORKSPACE`